azblob: fails to build on Dragonfly

Even in [v0.30.0](https://github.com/google/go-cloud/releases/tag/v0.30.0)

See #3261

So, the errors are the same

```
# github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/blockblob
../gomodcache/github.com/!azure/azure-sdk-for-go/sdk/storage/azblob@v1.0.0/blockblob/chunkwriting.go:273:15: undefined: mmb
../gomodcache/github.com/!azure/azure-sdk-for-go/sdk/storage/azblob@v1.0.0/blockblob/chunkwriting.go:279:65: undefined: mmb
../gomodcache/github.com/!azure/azure-sdk-for-go/sdk/storage/azblob@v1.0.0/blockblob/chunkwriting.go:281:22: undefined: mmb
../gomodcache/github.com/!azure/azure-sdk-for-go/sdk/storage/azblob@v1.0.0/blockblob/chunkwriting.go:287:39: undefined: mmb
../gomodcache/github.com/!azure/azure-sdk-for-go/sdk/storage/azblob@v1.0.0/blockblob/chunkwriting.go:293:18: undefined: newMMB
../gomodcache/github.com/!azure/azure-sdk-for-go/sdk/storage/azblob@v1.0.0/blockblob/chunkwriting.go:303:37: undefined: mmb
• core: Total in 127.46s …
error running command: exit status 1

Exited with code exit status 1
```

When I said "fixed upstream" I meant that the issue I opened in the Azure repo above was closed as fixed with a commit, but I guess that fix hasn't made it into a release yet (I haven't checked).

So, the more general problem is that we (that includes me) don't test these build tags as part of the daily CI builds, so we end up getting these errors very late when we do a binary release. I will try to fix this on my end so I don't have to "revert go-cloud" so frequently.

Can you point me to the Hugo smoke tests that manage that?

I tried this locally and got an error but a different one than you quoted:

```
$ cd blob/azureblob
$ GOOS=dragonfly go build
# runtime
/usr/lib/google-golang/src/runtime/googleexit.go:81:6: exit defined in both Go and assembly
```

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -42,3 +42,10 @@ jobs:
       - name: Run Tests
         shell: bash
         run: 'internal/testing/runchecks.sh'
+      - if: matrix.os == 'ubuntu-latest'
+        name: Build for dragonfly
+        run: |
+          go install ./...
+        env:
+          GOARCH: amd64
+          GOOS: dragonfly

